autoreduce_qp/model/database/records.py

```diff
@@ -66,10 +68,30 @@ def get_script_and_arguments(instrument: Instrument, script: str, arguments: dic
     if not arguments:
         arguments = VariableUtils.get_default_variables(instrument)
 
+    fetch_from_remote_source(arguments)
     arguments_str = json.dumps(arguments, separators=(',', ':'))
     return script, arguments_str
 
 
+def fetch_from_remote_source(arguments):
+    """
+    Update a supplied dictionary to contain the texts within any supplied raw
+    GitHub files.
+    """
+    for k1, v1 in arguments.items():
+        for k2 in v1:
+            nested_value = arguments[k1][k2]
+            if isinstance(nested_value, dict):
+                if "url" in nested_value and "default" in nested_value:
+                    url = nested_value["url"] + nested_value["default"]
+
+                    req = requests.get(url)
```

Comment:
This should be in a try: ... except..., check out [how these are handled in the webapp](https://github.com/ISISScientificComputing/autoreduce-frontend/blob/master/autoreduce_frontend/reduction_viewer/views/configure_new_batch_run.py#L76-L88) though we don't want a similar `render_error` call

Also note that for the `ConnectionError` in the example to work you will need to import `from requests.exceptions import ConnectionError` because for whatever silly reason that is NOT the same as the `__builtins__.ConnectionError`...